### ISSUE TITLE
RLlib: agent.restore(checkpoint) TypeError: can't convert np.ndarray of type numpy.object_.

### ISSUE THREAD

**Issue by dejangrubisic** [ray-project/ray#27262]
### What happened + What you expected to happen

I wanted to implement simple RLlib training with torch and PPO training. The documentation says this is the way we do it:

```
config = {"env": env_name, "framework": torch}
analysis = ray.tune.run(ppo.PPOTrainer, config=config)
checkpoint_path = analysis.get_best_checkpoint(trial=analysis.get_best_trial())
agent = ppo.PPOTrainer(env=env_name, config=config)
agent.restore(checkpoint_path) <<<< This fails with
```

`*** TypeError: can't convert np.ndarray of type numpy.object_. The only supported types are: float64, float32, float16, complex64, complex128, int64, int32, int16, int8, uint8, and bool.`

Note: 
This doesn't fail with Tensorflow

### Versions / Dependencies

Versions:

- ray == 1.13.0
- torch == 1.12.0
- Python == 3.8.13
- OS == Ubuntu 20.04.2 LTS (Focal Fossa)

### Reproduction script

```
# Run example: python sb2rllib_rllib_example.py

import gym
import ray
import ray.rllib.agents.ppo as ppo

# settings used for both stable baselines and rllib
env_name = "CartPole-v1"
train_steps = 10000
learning_rate = 1e-3
save_dir = "saved_models"
config = {
        "env": env_name, 
        "lr": learning_rate,
        "num_gpus": 0,
        "framework": "torch",
        }

# training and saving
analysis = ray.tune.run(
    "PPO",
    stop={"timesteps_total": train_steps},
    config=config,
    checkpoint_at_end=True,
    local_dir=save_dir,
)
# retrieve the checkpoint path
analysis.default_metric = "episode_reward_mean"
analysis.default_mode = "max"
checkpoint_path = analysis.get_best_checkpoint(trial=analysis.get_best_trial())
print(f"Trained model saved at {checkpoint_path}")

# load and restore model
agent = ppo.PPOTrainer(env=env_name, config=config)
agent.restore(checkpoint_path)
print(f"Agent loaded from saved model at {checkpoint_path}")
```

### Issue Severity

High: It blocks me from completing my task.

**Comment by Rohan138** [comment]
This is fixed by #26863; installing the latest nightly build following [this](https://docs.ray.io/en/latest/ray-overview/installation.html), or downgrading to `torch<1.12.0` should work.

**Comment by dejangrubisic** [comment]
Thanks a lot! I can confirm that
`pip install -U https://s3-us-west-2.amazonaws.com/ray-wheels/latest/ray-3.0.0.dev0-cp38-cp38-manylinux2014_x86_64.whl`

solved the problem :)
